## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [2]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [4]:
# Write your code for importing the csv file here
movies = pd.read_csv(r'C:\Users\prdahiya\OneDrive - Nokia\Upgrad\MS\movies.csv')
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [5]:
# Write your code for inspection here
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

In [6]:
movies.describe()

num_critic_for_reviews     duration  director_facebook_likes  \
count             3852.000000  3852.000000              3853.000000   
mean               163.046989   109.921080               784.507397   
std                123.927258    22.741728              3027.395958   
min                  1.000000    34.000000                 0.000000   
25%                 72.000000    95.000000                10.000000   
50%                134.000000   106.000000                58.000000   
75%                221.000000   120.000000               222.000000   
max                813.000000   330.000000             23000.000000   

       actor_3_facebook_likes  actor_1_facebook_likes         gross  \
count             3847.000000             3853.000000  3.853000e+03   
mean               747.290096             7575.558526  5.096547e+07   
std               1841.623945            15405.383114  6.932033e+07   
min                  0.000000                0.000000  1.620000e+02   
25%                183.000000              721.000000  6.830957e+06   
50%                427.000000             1000.000000  2.790000e+07   
75%                685.000000            12000.000000  6.550000e+07   
max              23000.000000           640000.000000  7.605058e+08   

       num_voted_users  cast_total_facebook_likes  facenumber_in_poster  \
count     3.853000e+03                3853.000000           3847.000000   
mean      1.024240e+05               11239.480145              1.378217   
std       1.502942e+05               18922.948757              2.056071   
min       5.000000e+00                   0.000000              0.000000   
25%       1.730900e+04                1817.000000              0.000000   
50%       5.052300e+04                3876.000000              1.000000   
75%       1.241850e+05               15972.000000              2.000000   
max       1.689764e+06              656730.000000             43.000000   

       num_user_for_reviews        budget   title_year  \
count           3853.000000  3.853000e+03  3853.000000   
mean             326.720478  4.524817e+07  2003.061511   
std              407.920904  2.234208e+08    10.007168   
min                1.000000  2.180000e+02  1920.000000   
25%              102.000000  1.000000e+07  1999.000000   
50%              203.000000  2.400000e+07  2005.000000   
75%              391.000000  5.000000e+07  2010.000000   
max             5060.000000  1.221550e+10  2016.000000   

       actor_2_facebook_likes   imdb_score  aspect_ratio  movie_facebook_likes  
count             3852.000000  3853.000000   3781.000000           3853.000000  
mean              1959.018432     6.462886      2.109569           9089.546068  
std               4472.171290     1.053843      0.353236          21277.375223  
min                  0.000000     1.600000      1.180000              0.000000  
25%                362.000000     5.900000      1.850000              0.000000  
50%                664.000000     6.600000      2.350000            206.000000  
75%                971.000000     7.200000      2.350000          11000.000000  
max             137000.000000     9.300000     16.000000         349000.000000

In [7]:
movies.shape

(3853, 28)

In [8]:
movies.size

107884

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [17]:
d=movies.isna().sum()
j=pd.DataFrame(d)
j[j[0]!=0].count()

0    12
dtype: int64

In [21]:
(movies.isnull().sum()>0).sum()

12

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [22]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations

movies.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [23]:
m = movies.drop(['color',
'director_facebook_likes',
'actor_1_facebook_likes',
'actor_2_facebook_likes',
'actor_3_facebook_likes',
'actor_2_name',
'cast_total_facebook_likes',
'actor_3_name',
'duration',
'facenumber_in_poster',
'content_rating',
'country',
'movie_imdb_link',
'aspect_ratio',
'plot_keywords'],axis=1)
m

director_name  num_critic_for_reviews        gross  \
0         James Cameron                   723.0  760505847.0   
1        Gore Verbinski                   302.0  309404152.0   
2            Sam Mendes                   602.0  200074175.0   
3     Christopher Nolan                   813.0  448130642.0   
4        Andrew Stanton                   462.0   73058679.0   
...                 ...                     ...          ...   
3848      Shane Carruth                   143.0     424760.0   
3849   Neill Dela Llana                    35.0      70071.0   
3850   Robert Rodriguez                    56.0    2040920.0   
3851       Edward Burns                    14.0       4584.0   
3852           Jon Gunn                    43.0      85222.0   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language       budget  title_year  imdb_score  \
0                   3054.0  English  237000000.0      2009.0         7.9   
1                   1238.0  English  300000000.0      2007.0         7.1   
2                    994.0  English  245000000.0      2015.0         6.8   
3                   2701.0  English  250000000.0      2012.0         8.5   
4                    738.0  English  263700000.0      2012.0         6.6   
...                    ...      ...          ...         ...         ...   
3848                 371.0  English       7000.0      2004.0         7.0   
3849                  35.0  English       7000.0      2005.0         6.3   
3850                 130.0  Spanish       7000.0      1992.0         6.9   
3851                  14.0  English       9000.0      2011.0         6.4   
3852                  84.0  English       1100.0      2004.0         6.6   

      movie_facebook_likes  
0                    33000  
1                        0  
2                    85000  
3                   164000  
4                    24000  
...                    ...  
3848                 19000  
3849                    74  
3850                     0  
3851                   413  
3852                   456  

[3853 rows x 13 columns]

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [38]:
# Write you code here
(movies.isnull().sum()/movies.count()*100).sort_values(ascending=False)

aspect_ratio                 1.904258
content_rating               1.261498
plot_keywords                0.784724
actor_3_name                 0.155966
facenumber_in_poster         0.155966
actor_3_facebook_likes       0.155966
language                     0.077922
color                        0.051935
num_critic_for_reviews       0.025961
actor_2_facebook_likes       0.025961
duration                     0.025961
actor_2_name                 0.025961
gross                        0.000000
director_facebook_likes      0.000000
imdb_score                   0.000000
title_year                   0.000000
budget                       0.000000
country                      0.000000
num_user_for_reviews         0.000000
genres                       0.000000
movie_imdb_link              0.000000
actor_1_facebook_likes       0.000000
director_name                0.000000
cast_total_facebook_likes    0.000000
num_voted_users              0.000000
movie_title                  0.000000
actor_1_name

In [40]:
round(100*(movies.isnull().sum()/len(movies.index)), 2).sort_values(ascending=False)

aspect_ratio                 1.87
content_rating               1.25
plot_keywords                0.78
actor_3_name                 0.16
facenumber_in_poster         0.16
actor_3_facebook_likes       0.16
language                     0.08
color                        0.05
num_critic_for_reviews       0.03
actor_2_facebook_likes       0.03
duration                     0.03
actor_2_name                 0.03
gross                        0.00
director_facebook_likes      0.00
imdb_score                   0.00
title_year                   0.00
budget                       0.00
country                      0.00
num_user_for_reviews         0.00
genres                       0.00
movie_imdb_link              0.00
actor_1_facebook_likes       0.00
director_name                0.00
cast_total_facebook_likes    0.00
num_voted_users              0.00
movie_title                  0.00
actor_1_name                 0.00
movie_facebook_likes         0.00
dtype: float64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [44]:
# Write your code for filling the NaN values in the 'language' column here
movies['language'].value_counts()

English       3671
French          37
Spanish         26
Mandarin        14
German          13
Japanese        12
Hindi           10
Cantonese        8
Italian          7
Portuguese       5
Korean           5
Norwegian        4
Dutch            3
Persian          3
Thai             3
Danish           3
Aboriginal       2
Dari             2
Indonesian       2
Hebrew           2
Russian          1
Romanian         1
Vietnamese       1
Arabic           1
Dzongkha         1
Zulu             1
None             1
Aramaic          1
Czech            1
Telugu           1
Icelandic        1
Filipino         1
Hungarian        1
Maya             1
Bosnian          1
Mongolian        1
Kazakh           1
Swedish          1
Name: language, dtype: int64

In [46]:
movies['language'].isnull().sum()

3

In [47]:
movies.loc[pd.isnull(movies['language']), ['language']] = 'English'

In [48]:
(movies.language == 'English').sum()

3674

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [58]:
# Write your code for unit conversion here
movies['Budget in millions']=movies['budget']/1000000
movies['Gross in millions']=movies['gross']/1000000
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ...      country content_rating  \
0         Action|Adventure|Fantasy|Sci-Fi  ...          USA          PG-13   
1                Action|Adventure|Fantasy  ...          USA          PG-13   
2               Action|Adventure|Thriller  ...           UK          PG-13   
3                         Action|Thriller  ...          USA          PG-13   
4                 Action|Adventure|Sci-Fi  ...          USA          PG-13   
...                                   ...  ...          ...            ...   
3848                Drama|Sci-Fi|Thriller  ...          USA          PG-13   
3849                             Thriller  ...  Philippines      Not Rated   
3850  Action|Crime|Drama|Romance|Thriller  ...          USA              R   
3851                         Comedy|Drama  ...          USA      Not Rated   
3852                          Documentary  ...          USA             PG   

           budget  title_year actor_2_facebook_likes  imdb_score aspect_ratio  \
0     237000000.0      2009.0                  936.0         7.9         1.78   
1     300000000.0      2007.0                 5000.0         7.1         2.35   
2     245000000.0      2015.0                  393.0         6.8         2.35   
3     250000000.0      2012.0                23000.0         8.5         2.35   
4     263700000.0      2012.0                  632.0         6.6         2.35   
...           ...         ...                    ...         ...          ...   
3848       7000.0      2004.0                   45.0         7.0         1.85   
3849       7000.0      2005.0                    0.0         6.3          NaN   
3850       7000.0      1992.0                   20.0     

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [62]:
# Write your code for creating the profit column here
movies['profit']=movies['gross']-movies['budget']
movies.sort_values(by='profit',ascending=False)

color     director_name  num_critic_for_reviews  duration  \
0     Color     James Cameron                   723.0     178.0   
28    Color   Colin Trevorrow                   644.0     124.0   
25    Color     James Cameron                   315.0     194.0   
2704  Color      George Lucas                   282.0     125.0   
2748  Color  Steven Spielberg                   215.0     120.0   
...     ...               ...                     ...       ...   
2147  Color   Katsuhiro Ôtomo                   105.0     103.0   
2136  Color    Hayao Miyazaki                   174.0     134.0   
2693  Color      Lajos Koltai                    73.0     134.0   
3282  Color    Chan-wook Park                   202.0     112.0   
2678  Color      Joon-ho Bong                   363.0     110.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
28                      365.0                  1000.0          Judy Greer   
25                        0.0                   794.0        Kate Winslet   
2704                      0.0                   504.0       Peter Cushing   
2748                  14000.0                   548.0         Dee Wallace   
...                       ...                     ...                 ...   
2147                     78.0                   101.0  Robin Atkin Downes   
2136                   6000.0                   745.0  Jada Pinkett Smith   
2693                     45.0                     0.0     Péter Fancsikai   
3282                      0.0                    38.0        Yeong-ae Lee   
2678                    584.0                    74.0        Kang-ho Song   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
28                    3000.0  652177271.0   
25                   29000.0  658672302.0   
2704                 11000.0  460935665.0   
2748                   861.0  434949459.0   
...                      ...          ...   
2147                   488.0     410388.0   
2136                   893.0    2298191.0   
2693                     9.0     195888.0   
3282                   717.0     211667.0   
2678                   629.0    2201412.0   

                                                 genres  ... content_rating  \
0                       Action|Adventure|Fantasy|Sci-Fi  ...          PG-13   
28                     Action|Adventure|Sci-Fi|Thriller  ...          PG-13   
25                                        Drama|Romance  ...          PG-13   
2704                    Action|Adventure|Fantasy|Sci-Fi  ...             PG   
2748                                      Family|Sci-Fi  ...             PG   
...                                                 ...  ...            ...   
2147  Action|Adventure|Animation|Family|Sci-Fi|Thriller  ...          PG-13   
2136                        Adventure|Animation|Fantasy  ...          PG-13   
2693                                  Drama|Romance|War  ...              R   
3282                                        Crime|Drama  ...              R   
2678                         Comedy|Drama|Horror|Sci-Fi  ...              R   

            budget  title_year  actor_2_facebook_likes imdb_score  \
0     2.370000e+08      2009.0                   936.0        7.9   
28    1.500000e+08      2015.0                  2000.0        7.0   
25    2.000000e+08      1997.0                 14000.0        7.7   
2704  1.100000e+07      1977.0                  1000.0        8.7   
2748  1.050000e+07      1982.0                   725.0        7.9   
...            ...         ...                     ...        ...   
2147  2.127520e+09      2004.0                   336.0        6.9   
2136  2.400000e+09      1997.0                   851.0        8.4   
2693  2.500000e+09      2005.0                     2.0        7.1   
3282  4.200000e+09      2005.0                   126.0        7.7   
2678  1.221550e+10      200

In [ ]:
# Write your code for sorting the dataframe here


In [68]:
 # Write your code to get the top 10 profiting movies here
top10 = movies.sort_values(by='profit',ascending=False).head(10)
top10[['director_name','movie_title']]

director_name                                 movie_title
0         James Cameron                                     Avatar 
28      Colin Trevorrow                             Jurassic World 
25        James Cameron                                    Titanic 
2704       George Lucas         Star Wars: Episode IV - A New Hope 
2748   Steven Spielberg                 E.T. the Extra-Terrestrial 
16          Joss Whedon                               The Avengers 
482        Roger Allers                              The Lion King 
230        George Lucas  Star Wars: Episode I - The Phantom Menace 
64    Christopher Nolan                            The Dark Knight 
419           Gary Ross                           The Hunger Games

In [65]:
movies2 = pd.read_csv(r'C:\Users\prdahiya\OneDrive - Nokia\Upgrad\MS\movies.csv')
movies2

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

In [66]:
movies2.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [70]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies.sort_values(by='imdb_score',ascending=False).head(250)
IMDb_Top_250

color         director_name  num_critic_for_reviews  duration  \
1795  Color        Frank Darabont                   199.0     142.0   
3016  Color  Francis Ford Coppola                   208.0     175.0   
2543  Color  Francis Ford Coppola                   149.0     220.0   
64    Color     Christopher Nolan                   645.0     152.0   
325   Color         Peter Jackson                   328.0     192.0   
...     ...                   ...                     ...       ...   
677   Color            Doug Liman                   249.0     119.0   
1734  Color          Pierre Morel                   309.0      93.0   
86    Color              Don Hall                   384.0     102.0   
746   Color         Cameron Crowe                   149.0     152.0   
157   Color          Dean DeBlois                   292.0     102.0   

      director_facebook_likes  actor_3_facebook_likes     actor_2_name  \
1795                      0.0                   461.0   Jeffrey DeMunn   
3016                      0.0                  3000.0    Marlon Brando   
2543                      0.0                  3000.0        Al Pacino   
64                    22000.0                 11000.0     Heath Ledger   
325                       0.0                   416.0       Billy Boyd   
...                       ...                     ...              ...   
677                     218.0                    73.0    Josh Hamilton   
1734                    180.0                   485.0    Holly Valance   
86                       38.0                   562.0    Daniel Henney   
746                     488.0                   947.0  Zooey Deschanel   
157                     255.0                   953.0   Djimon Hounsou   

      actor_1_facebook_likes        gross  \
1795                 11000.0   28341469.0   
3016                 14000.0  134821952.0   
2543                 22000.0   57300000.0   
64                   23000.0  533316061.0   
325                   5000.0  377019252.0   
...                      ...          ...   
677                  13000.0  121468960.0   
1734                 14000.0  145000989.0   
86                     756.0  222487711.0   
746                  22000.0   32522352.0   
157                  18000.0  176997107.0   

                                                 genres  ... content_rating  \
1795                                        Crime|Drama  ...              R   
3016                                        Crime|Drama  ...              R   
2543                                        Crime|Drama  ...              R   
64                          Action|Crime|Drama|Thriller  ...          PG-13   
325                      Action|Adventure|Drama|Fantasy  ...          PG-13   
...                                                 ...  ...            ...   
677                             Action|Mystery|Thriller  ...          PG-13   
1734                                    Action|Thriller  ...          PG-13   
86    Action|Adventure|Animation|Comedy|Drama|Family...  ...             PG   
746                        Adventure|Comedy|Drama|Music  ...              R   
157    Action|Adventure|Animation|Comedy|Family|Fantasy  ...             PG   

           budget  title_year  actor_2_facebook_likes imdb_score  \
1795   25000000.0      1994.0                   745.0        9.3   
3016    6000000.0      1972.0                 10000.0        9.2   
2543   13000000.0      1974.0                 14000.0        9.0   
64    185000000.0      2008.0                 13000.0        9.0   
325    94000000.0      2003.0                   857.0        8.9   
...           ...         ...                     ...        ...   
677    60000000.0      2002.0                   147.0        7.9   
1734   25000000.0      2008.0                   816.0        7.9   
86    165000000.0      2014.0                   719.0        7.9   
746    60000000.0      2000.0                 11000.0        7.9   
157   145000000.0      201

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

In [76]:
movies.loc[(movies['imdb_score']>=7.5) & (movies['imdb_score']<8),'imdb_bucket']='7.5 to 8'
movies.loc[(movies['imdb_score']>=8) & (movies['imdb_score']<8.5),'imdb_bucket']='8 to 8.5'
movies.loc[(movies['imdb_score']>=8.5) & (movies['imdb_score']<9),'imdb_bucket']='8.5 to 9'
movies.loc[(movies['imdb_score']>=9) & (movies['imdb_score']<9.5),'imdb_bucket']='9 to 9.5'
movies.loc[(movies['imdb_score']>=9.5) & (movies['imdb_score']<10),'imdb_bucket']='9.5 to 10'
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ...       budget title_year  \
0         Action|Adventure|Fantasy|Sci-Fi  ...  237000000.0     2009.0   
1                Action|Adventure|Fantasy  ...  300000000.0     2007.0   
2               Action|Adventure|Thriller  ...  245000000.0     2015.0   
3                         Action|Thriller  ...  250000000.0     2012.0   
4                 Action|Adventure|Sci-Fi  ...  263700000.0     2012.0   
...                                   ...  ...          ...        ...   
3848                Drama|Sci-Fi|Thriller  ...       7000.0     2004.0   
3849                             Thriller  ...       7000.0     2005.0   
3850  Action|Crime|Drama|Romance|Thriller  ...       7000.0     1992.0   
3851                         Comedy|Drama  ...       9000.0     2011.0   
3852                          Documentary  ...       1100.0     2004.0   

      actor_2_facebook_likes  imdb_score aspect_ratio  movie_facebook_likes  \
0                      936.0         7.9         1.78                 33000   
1                     5000.0         7.1         2.35                     0   
2                      393.0         6.8         2.35                 85000   
3                    23000.0         8.5         2.35                164000   
4                      632.0         6.6         2.35                 24000   
...                      ...         ...          ...                   ...   
3848                    45.0         7.0         1.85                 19000   
3849                     0.0         6.3          NaN                    74   
3850                    20.0         6.9         1.37                     0   
3851                   205.0         6.4    

In [77]:
movies['imdb_bucket'].value_counts()

7.5 to 8    3638
8 to 8.5     168
8.5 to 9      43
9 to 9.5       4
Name: imdb_bucket, dtype: int64

In [79]:
IMDb_Top_250.loc[(IMDb_Top_250['imdb_score']>=7.5) & (IMDb_Top_250['imdb_score']<8),'imdb_bucket']='7.5 to 8'
IMDb_Top_250.loc[(IMDb_Top_250['imdb_score']>=8) & (IMDb_Top_250['imdb_score']<8.5),'imdb_bucket']='8 to 8.5'
IMDb_Top_250.loc[(IMDb_Top_250['imdb_score']>=8.5) & (IMDb_Top_250['imdb_score']<9),'imdb_bucket']='8.5 to 9'
IMDb_Top_250.loc[(IMDb_Top_250['imdb_score']>=9) & (IMDb_Top_250['imdb_score']<9.5),'imdb_bucket']='9 to 9.5'
IMDb_Top_250.loc[(IMDb_Top_250['imdb_score']>=9.5) & (IMDb_Top_250['imdb_score']<10),'imdb_bucket']='9.5 to 10'
IMDb_Top_250

color         director_name  num_critic_for_reviews  duration  \
1795  Color        Frank Darabont                   199.0     142.0   
3016  Color  Francis Ford Coppola                   208.0     175.0   
2543  Color  Francis Ford Coppola                   149.0     220.0   
64    Color     Christopher Nolan                   645.0     152.0   
325   Color         Peter Jackson                   328.0     192.0   
...     ...                   ...                     ...       ...   
677   Color            Doug Liman                   249.0     119.0   
1734  Color          Pierre Morel                   309.0      93.0   
86    Color              Don Hall                   384.0     102.0   
746   Color         Cameron Crowe                   149.0     152.0   
157   Color          Dean DeBlois                   292.0     102.0   

      director_facebook_likes  actor_3_facebook_likes     actor_2_name  \
1795                      0.0                   461.0   Jeffrey DeMunn   
3016                      0.0                  3000.0    Marlon Brando   
2543                      0.0                  3000.0        Al Pacino   
64                    22000.0                 11000.0     Heath Ledger   
325                       0.0                   416.0       Billy Boyd   
...                       ...                     ...              ...   
677                     218.0                    73.0    Josh Hamilton   
1734                    180.0                   485.0    Holly Valance   
86                       38.0                   562.0    Daniel Henney   
746                     488.0                   947.0  Zooey Deschanel   
157                     255.0                   953.0   Djimon Hounsou   

      actor_1_facebook_likes        gross  \
1795                 11000.0   28341469.0   
3016                 14000.0  134821952.0   
2543                 22000.0   57300000.0   
64                   23000.0  533316061.0   
325                   5000.0  377019252.0   
...                      ...          ...   
677                  13000.0  121468960.0   
1734                 14000.0  145000989.0   
86                     756.0  222487711.0   
746                  22000.0   32522352.0   
157                  18000.0  176997107.0   

                                                 genres  ...       budget  \
1795                                        Crime|Drama  ...   25000000.0   
3016                                        Crime|Drama  ...    6000000.0   
2543                                        Crime|Drama  ...   13000000.0   
64                          Action|Crime|Drama|Thriller  ...  185000000.0   
325                      Action|Adventure|Drama|Fantasy  ...   94000000.0   
...                                                 ...  ...          ...   
677                             Action|Mystery|Thriller  ...   60000000.0   
1734                                    Action|Thriller  ...   25000000.0   
86    Action|Adventure|Animation|Comedy|Drama|Family...  ...  165000000.0   
746                        Adventure|Comedy|Drama|Music  ...   60000000.0   
157    Action|Adventure|Animation|Comedy|Family|Fantasy  ...  145000000.0   

     title_year  actor_2_facebook_likes  imdb_score aspect_ratio  \
1795     1994.0                   745.0         9.3         1.85   
3016     1972.0                 10000.0         9.2         1.85   
2543     1974.0                 14000.0         9.0         1.85   
64       2008.0                 13000.0         9.0         2.35   
325      2003.0                   857.0         8.9         2.35   
...         ...                     ...         ...          ...   
677      2002.0                   147.0         7.9         2.35   
1734     2008.0                   816.0         7.9         2.35   
86       2014.0                   719.0         7.9         2.39   
746      2000.0                 11000.0         7.9         1.85   
157      2014.0                  3000.0         7.

In [80]:
IMDb_Top_250['imdb_bucket'].value_counts()

8 to 8.5    168
8.5 to 9     43
7.5 to 8     35
9 to 9.5      4
Name: imdb_bucket, dtype: int64

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [84]:
# Write your code for creating three new dataframes here
# Include all movies in which Meryl_Streep is the lead
Meryl_Streep = movies[(movies['actor_1_name']=='Meryl Streep')]
Meryl_Streep

color   director_name  num_critic_for_reviews  duration  \
392   Color    Nancy Meyers                   187.0     120.0   
1038  Color   Curtis Hanson                    42.0     111.0   
1132  Color     Nora Ephron                   252.0     123.0   
1322  Color   David Frankel                   208.0     109.0   
1390  Color  Robert Redford                   227.0      92.0   
1471  Color  Sydney Pollack                    66.0     161.0   
1514  Color   David Frankel                   234.0     100.0   
1563  Color   Carl Franklin                    64.0     127.0   
1784  Color  Stephen Daldry                   174.0     114.0   
2500  Color  Phyllida Lloyd                   331.0     105.0   
2793  Color   Robert Altman                   211.0     105.0   

      director_facebook_likes  actor_3_facebook_likes       actor_2_name  \
392                     278.0                   963.0     Hunter Parrish   
1038                    161.0                   132.0    Glenn Morshower   
1132                      0.0                   923.0  Mary Lynn Rajskub   
1322                     64.0                   505.0      Anne Hathaway   
1390                      0.0                 10000.0         Tom Cruise   
1471                    521.0                   184.0      Michael Gough   
1514                     64.0                   329.0       Steve Carell   
1563                     73.0                   433.0       William Hurt   
1784                    335.0                   530.0    Stephen Dillane   
2500                     58.0                   583.0      Jim Broadbent   
2793                    500.0                   718.0    Virginia Madsen   

      actor_1_facebook_likes        gross                           genres  \
392                  11000.0  112703470.0             Comedy|Drama|Romance   
1038                 11000.0   46815748.0  Action|Adventure|Crime|Thriller   
1132                 11000.0   94125426.0          Biography|Drama|Romance   
1322                 11000.0  124732962.0             Comedy|Drama|Romance   
1390                 11000.0   14998070.0               Drama|Thriller|War   
1471                 11000.0   87100000.0          Biography|Drama|Romance   
1514                 11000.0   63536011.0             Comedy|Drama|Romance   
1563                 11000.0   23209440.0                            Drama   
1784                 11000.0   41597830.0                    Drama|Romance   
2500                 11000.0   29959436.0          Biography|Drama|History   
2793                 11000.0   20338609.0               Comedy|Drama|Music   

      ...      budget title_year  actor_2_facebook_likes  imdb_score  \
392   ...  85000000.0     2009.0                  2000.0         6.6   
1038  ...  45000000.0     1994.0                   894.0         6.3   
1132  ...  40000000.0     2009.0                   935.0         7.0   
1322  ...  35000000.0     2006.0                 11000.0         6.8   
1390  ...  35000000.0     2007.0                 10000.0         6.2   
1471  ...  31000000.0     1985.0                   920.0         7.2   
1514  ...  30000000.0     2012.0                  7000.0         6.3   
1563  ...  30000000.0     1998.0                   882.0         7.0   
1784  ...  25000000.0     2002.0                   577.0         7.6   
2500  ...  13000000.0     2011.0                  1000.0         6.4   
2793  ...  10000000.0     2006.0                   912.0         6.8   

     aspect_ratio  movie_facebook_likes Budget in millions Gross in millions  \
392          1.85                     0               85.0        112.703470   
1038         2.35                     0               45.0         46.815748   
1132         1.85                 13000               40.0         94.125426   
1322         2.35                     0               35.0        124.732962   
1390         2.35                     0               35.0         14.998070   
1471         1.85             

In [85]:
# Include all movies in which Leo_Caprio is the lead
Leo_Caprio = movies[(movies['actor_1_name']=='Leonardo DiCaprio')]
Leo_Caprio

color          director_name  num_critic_for_reviews  \
25               Color          James Cameron                   315.0   
49               Color           Baz Luhrmann                   490.0   
94               Color      Christopher Nolan                   642.0   
173              Color  Alejandro G. Iñárritu                   556.0   
246    Black and White        Martin Scorsese                   267.0   
283              Color      Quentin Tarantino                   765.0   
293              Color           Edward Zwick                   166.0   
294              Color        Martin Scorsese                   606.0   
312              Color        Martin Scorsese                   233.0   
347              Color        Martin Scorsese                   352.0   
431              Color        Martin Scorsese                   490.0   
608              Color           Ridley Scott                   238.0   
859              Color       Steven Spielberg                   194.0   
934              Color            Danny Boyle                   118.0   
1046             Color             Sam Mendes                   323.0   
1336             Color        Randall Wallace                    83.0   
1364             Color         Clint Eastwood                   392.0   
1457   Black and White              Sam Raimi                    63.0   
1909             Color             Jerry Zaks                    45.0   
2484             Color           Baz Luhrmann                   106.0   
3025             Color           Baz Luhrmann                   490.0   

      duration  director_facebook_likes  actor_3_facebook_likes  \
25       194.0                      0.0                   794.0   
49       143.0                   1000.0                    77.0   
94       148.0                  22000.0                 23000.0   
173      156.0                      0.0                   733.0   
246      170.0                  17000.0                   827.0   
283      165.0                  16000.0                   265.0   
293      143.0                    380.0                   452.0   
294      240.0                  17000.0                  4000.0   
312      216.0                  17000.0                  1000.0   
347      151.0                  17000.0                  1000.0   
431      138.0                  17000.0                   163.0   
608      128.0                      0.0                   135.0   
859      141.0                  14000.0                  3000.0   
934      119.0                      0.0                    39.0   
1046     119.0                      0.0                 10000.0   
1336     132.0                    130.0                    80.0   
1364     137.0                  16000.0                   363.0   
1457     107.0                      0.0                   159.0   
1909      98.0                      4.0                 11000.0   
2484     120.0                   1000.0                   954.0   
3025     143.0                   1000.0                    76.0   

             actor_2_name  actor_1_facebook_likes        gross  \
25           Kate Winslet                 29000.0  658672302.0   
49      Elizabeth Debicki                 29000.0  144812796.0   
94              Tom Hardy                 29000.0  292568851.0   
173             Tom Hardy                 29000.0  183635922.0   
246            Adam Scott                 29000.0  102608827.0   
283       Christoph Waltz                 29000.0  162804648.0   
293        Djimon Hounsou                 29000.0   57366262.0   
294   Matthew McConaughey                 29000.0  116866727.0   
312           Liam Neeson                 29000.0   77679638.0   
347            Matt Damon                 29000.0  132373442.0   
431         Joseph Sikora                 29000.0  127968405.0   
608        Simon McBurney                 29000.0   39380442.0   
859             Tom Hanks                 29000.0  164435221.0   
934      Vi

In [87]:
# Include all movies in which Brad_Pitt is the lead
Brad_Pitt = movies[(movies['actor_1_name']=='Brad Pitt')]
Brad_Pitt

color          director_name  num_critic_for_reviews  duration  \
97    Color          David Fincher                   362.0     166.0   
142   Color      Wolfgang Petersen                   220.0     196.0   
243   Color      Steven Soderbergh                   198.0     125.0   
244   Color             Doug Liman                   233.0     126.0   
367   Color             Tony Scott                   142.0     114.0   
383   Color      Steven Soderbergh                   186.0     116.0   
448   Color             David Ayer                   406.0     134.0   
579   Color    Jean-Jacques Annaud                    76.0     136.0   
646   Color          David Fincher                   315.0     151.0   
749   Color        Patrick Gilmore                    98.0      85.0   
887   Color            Neil Jordan                   120.0     123.0   
1397  Color        Terrence Malick                   584.0     139.0   
1608  Color         Andrew Dominik                   273.0     160.0   
2030  Color  Alejandro G. Iñárritu                   285.0     143.0   
2146  Color    Angelina Jolie Pitt                   131.0     122.0   
2428  Color         Andrew Dominik                   414.0      97.0   
2602  Color             Tony Scott                   122.0     121.0   

      director_facebook_likes  actor_3_facebook_likes         actor_2_name  \
97                    21000.0                   919.0        Jason Flemyng   
142                     249.0                   844.0        Orlando Bloom   
243                       0.0                   350.0        Julia Roberts   
244                     218.0                   322.0  Angelina Jolie Pitt   
367                   12000.0                   307.0      Stephen Dillane   
383                       0.0                   471.0           Bernie Mac   
448                     452.0                   697.0         Logan Lerman   
579                     218.0                   400.0                 Mako   
646                   21000.0                   637.0            Meat Loaf   
749                       0.0                    82.0     Adriano Giannini   
887                     277.0                  4000.0           Tom Cruise   
1397                      0.0                   687.0         Tye Sheridan   
1608                    181.0                   820.0        Jeremy Renner   
2030                      0.0                    51.0       Harriet Walter   
2146                  11000.0                   188.0  Angelina Jolie Pitt   
2428                    181.0                   748.0          Sam Shepard   
2602                  12000.0                   975.0          Gary Oldman   

      actor_1_facebook_likes        gross  \
97                   11000.0  127490802.0   
142                  11000.0  133228348.0   
243                  11000.0  125531634.0   
244                  11000.0  186336103.0   
367                  11000.0      26871.0   
383                  11000.0  183405771.0   
448                  11000.0   85707116.0   
579                  11000.0   37901509.0   
646                  11000.0   37023395.0   
749                  11000.0   26288320.0   
887                  11000.0  105264608.0   
1397                 11000.0   13303319.0   
1608                 11000.0    3904982.0   
2030                 11000.0   34300771.0   
2146                 11000.0     531009.0   
2428                 11000.0   14938570.0   
2602                 11000.0   12281500.0   

                                                 genres  ...       budget  \
97                                Drama|Fantasy|Romance  ...  150000000.0   
142                                           Adventure  ...  175000000.0   
243                                      Crime|Thriller  ...  110000000.0   
244                Action|Comedy|Crime|Romance|Thriller  ...  120000000.0   
367                               Action|Crime|Thriller  ...   92000000.0   
383                            

In [95]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep,Leo_Caprio,Brad_Pitt])
Combined

color          director_name  num_critic_for_reviews  \
392              Color           Nancy Meyers                   187.0   
1038             Color          Curtis Hanson                    42.0   
1132             Color            Nora Ephron                   252.0   
1322             Color          David Frankel                   208.0   
1390             Color         Robert Redford                   227.0   
1471             Color         Sydney Pollack                    66.0   
1514             Color          David Frankel                   234.0   
1563             Color          Carl Franklin                    64.0   
1784             Color         Stephen Daldry                   174.0   
2500             Color         Phyllida Lloyd                   331.0   
2793             Color          Robert Altman                   211.0   
25               Color          James Cameron                   315.0   
49               Color           Baz Luhrmann                   490.0   
94               Color      Christopher Nolan                   642.0   
173              Color  Alejandro G. Iñárritu                   556.0   
246    Black and White        Martin Scorsese                   267.0   
283              Color      Quentin Tarantino                   765.0   
293              Color           Edward Zwick                   166.0   
294              Color        Martin Scorsese                   606.0   
312              Color        Martin Scorsese                   233.0   
347              Color        Martin Scorsese                   352.0   
431              Color        Martin Scorsese                   490.0   
608              Color           Ridley Scott                   238.0   
859              Color       Steven Spielberg                   194.0   
934              Color            Danny Boyle                   118.0   
1046             Color             Sam Mendes                   323.0   
1336             Color        Randall Wallace                    83.0   
1364             Color         Clint Eastwood                   392.0   
1457   Black and White              Sam Raimi                    63.0   
1909             Color             Jerry Zaks                    45.0   
2484             Color           Baz Luhrmann                   106.0   
3025             Color           Baz Luhrmann                   490.0   
97               Color          David Fincher                   362.0   
142              Color      Wolfgang Petersen                   220.0   
243              Color      Steven Soderbergh                   198.0   
244              Color             Doug Liman                   233.0   
367              Color             Tony Scott                   142.0   
383              Color      Steven Soderbergh                   186.0   
448              Color             David Ayer                   406.0   
579              Color    Jean-Jacques Annaud                    76.0   
646              Color          David Fincher                   315.0   
749              Color        Patrick Gilmore                    98.0   
887              Color            Neil Jordan                   120.0   
1397             Color        Terrence Malick                   584.0   
1608             Color         Andrew Dominik                   273.0   
2030             Color  Alejandro G. Iñárritu                   285.0   
2146             Color    Angelina Jolie Pitt                   131.0   
2428             Color         Andrew Dominik                   414.0   
2602             Color             Tony Scott                   122.0   

      duration  director_facebook_likes  actor_3_facebook_likes  \
392      120.0                    278.0                   963.0   
1038     111.0                    161.0                   132.0   
1132     123.0                      0.0                   923.0   
1322     109.0                     64.0                   505.0   
1390      92.0                 

In [ ]:
# Write your code for grouping the combined dataframe here


In [101]:
# Write the code for finding the mean of critic reviews and audience reviews here
Combined['actor_1_name'].value_counts()

Leonardo DiCaprio    21
Brad Pitt            17
Meryl Streep         11
Name: actor_1_name, dtype: int64

#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [106]:
Combined_by_segment = Combined.groupby('actor_1_name')
Combined_by_segment

In [103]:
Combined_by_segment['num_critic_for_reviews'].mean()

actor_1_name
Brad Pitt            245.000000
Leonardo DiCaprio    330.190476
Meryl Streep         181.454545
Name: num_critic_for_reviews, dtype: float64

In [104]:
Combined_by_segment['num_user_for_reviews'].mean()

actor_1_name
Brad Pitt            742.352941
Leonardo DiCaprio    914.476190
Meryl Streep         297.181818
Name: num_user_for_reviews, dtype: float64

In [107]:
Combined.groupby('actor_1_name')[['num_critic_for_reviews','num_user_for_reviews']].mean()

num_critic_for_reviews  num_user_for_reviews
actor_1_name                                                   
Brad Pitt                      245.000000            742.352941
Leonardo DiCaprio              330.190476            914.476190
Meryl Streep                   181.454545            297.181818